troi/tools/utils.py

```diff
@@ -18,3 +21,142 @@ def create_http_session():
+        return http
+
+    def create_playlist(self, title, sharing="public" ,track_ids=None, description=None):
+        url = f"{SOUNDCLOUD_URL}/playlists"
+        data = {
+            "playlist": {
+                "title": title,
+                "sharing": sharing
+            }
+        }
+        if description:
+            data["playlist"]["description"] = description
+        response = self.session.post(url, headers=self.headers, data=json.dumps(data))
+        return response.json()
+
+    def add_playlist_tracks(self, playlist_id, track_ids):
+        url = f"{SOUNDCLOUD_URL}/playlists/{playlist_id}"
+        data = {
+            "playlist": {
+                "tracks": [{"id": track_id} for track_id in track_ids]
+            }
+        }
+        response = self.session.put(url, headers=self.headers, data=json.dumps(data))
+        return response.json()
+
+    def update_playlist_details(self, playlist_id, title=None, description=None):
+        url = f"{SOUNDCLOUD_URL}/playlists/{playlist_id}"
+        data = {
+            "playlist": {}
+        }
+        if title:
+            data["playlist"]["title"] = title
+        if description:
+            data["playlist"]["description"] = description
+
+        response = self.session.put(url, headers=self.headers, data=json.dumps(data))
+        return response.json()
+    
+    def get_track_details(self, track_ids, **kwargs):
+        track_details = []
+
+        for track_id in track_ids:
+            url = f"{SOUNDCLOUD_URL}/tracks/{track_id}"
+            response = self.session.get(url, headers=self.headers, params=kwargs)
+
+            if response.status_code != 200:
+                track_details.append({
+                    "id": track_id,
+                    "title": None,
+                    "playable": False
+                })
+                continue
+
+            data = response.json()
+            while data.get("collection"):
+                track_data = data["collection"][0]
+                track_details.append({
+                    "id": track_data["id"],
+                    "title": track_data["title"],
+                    "playable": track_data.get("streamable", True)
+                })
+
+                next_href = data.get("next_href")
+                if not next_href:
+                    break
+
+                data = self.session.get(next_href, headers=self.headers).json()
+
+        return track_details
+
+    def get_playlist_tracks(self, playlist_id, **kwargs):
+        tracks = []
+        url = f"{SOUNDCLOUD_URL}/playlists/{playlist_id}/tracks"
+
+        while url:
+            response = self.session.get(url, headers=self.headers, params=kwargs)
+            response.raise_for_status()
+            data = response.json()
+            tracks.extend(data.get("collection", []))
+            url = data.get("next_href")
+
+        print(tracks)
+        return tracks
+
+    def get_unplayable_tracks(self, playlist_id, **kwargs):
```

Comment:
This function is almost similar to the get playlist tracks one, you can call that function here to get the tracks in the playlist and then filter the unstreamable ones and return them.